sources/device.cpp

```diff
@@ -0,0 +1,36 @@
+#include "../headers/device.h"
+
+namespace smarthome {
+  
+Device::Device(const std::string& name, bool state) {
```

Comment:
Across all your project you should try using initializer list. The difference might not be significant in your case, but it's a generally a good practice(and some times is mandatory).

Have a quick look on the differences between the two.